src/components/app/components/Navbar.js

```diff
@@ -35,9 +42,16 @@ const items = [
  * this is the the navigation bar at the top of the home page
 */
 class Navbar extends React.Component {
+
     logout = ({ key }) => {
-        const { removeCookie } = this.props.actions;
-        key === SIGN_OUT && removeCookie();
+        const { removeCookie, resetSigninState } = this.props.actions;
+        const { isLoggedIn } = this.props;
+
+        if (isLoggedIn && key === SIGN_OUT) {
+            removeCookie();
+            message.info(VALID_SIGNOUT, 5);
+            resetSigninState();
+        }
```

Comment:
so a slight optimization - you only want to initialize `removeCookie and resetSigninState` if `(isLoggedIn && key === SIGN_OUT)`. So this is a bit more optimal:
```
        ...
        if (isLoggedIn && key === SIGN_OUT) {
            const { removeCookie, resetSigninState } = this.props.actions;

            removeCookie();
            resetSigninState();
            message.info(VALID_SIGNOUT, 5);
        }
```